tests/test_auto.py

```diff
@@ -32,9 +33,14 @@
     PeftModelForSequenceClassification,
     PeftModelForTokenClassification,
 )
+from peft.utils import infer_device
 
 
 class PeftAutoModelTester(unittest.TestCase):
+    def setUp(self):
+        if infer_device() == "mps":
+            return pytest.skip("bfloat16 is not supported on MPS")
```

Comment:
Instead of skipping all the tests in this class, I wonder if it wouldn't be better to remove the `torch_dtype=torch.bfloat16` setting here. The tests aren't really about bf16. Maybe this was added for memory reasons, but we could still use float16 if that were the case.

Pinging @younesbelkada since he added the tests.

Reply:
Since we are not performing any inference here they can indeed be in float16 - @akx can you try that out and see if it works?

Reply:
Yeah, that works. I'll make it do `float16` on MPS and `bfloat16` otherwise.